picard/tagger.py

```diff
@@ -1086,29 +1099,45 @@ def main(localedir=None, autoupdate=True):
     signal.signal(signal.SIGINT, signal.SIG_DFL)
 
     picard_args, unparsed_args = process_picard_args()
+
     if picard_args.version:
         return version()
     if picard_args.long_version:
         return longversion()
 
-    try:
-        from PyQt5.QtDBus import QDBusConnection
-        dbus = QDBusConnection.sessionBus()
-        dbus.registerService(PICARD_APP_ID)
-    except ImportError:
-        pass
+    should_start = picard_args.stand_alone_instance
 
-    tagger = Tagger(picard_args, unparsed_args, localedir, autoupdate)
-
-    # Initialize Qt default translations
-    translator = QtCore.QTranslator()
-    locale = QtCore.QLocale()
-    translation_path = QtCore.QLibraryInfo.location(QtCore.QLibraryInfo.LibraryLocation.TranslationsPath)
-    log.debug("Looking for Qt locale %s in %s", locale.name(), translation_path)
-    if translator.load(locale, "qtbase_", directory=translation_path):
-        tagger.installTranslator(translator)
+    if not should_start:
+        pipe_handler = pipe.Pipe(app_name=PICARD_APP_NAME, app_version=PICARD_FANCY_VERSION_STR, args=picard_args.FILE)
+        should_start = True in [pipe_handler.permission_error_happened, pipe_handler.is_pipe_owner]
+        if pipe_handler.permission_error_happened:
+            pipe_handler = None
     else:
-        log.debug('Qt locale %s not available', locale.name())
+        pipe_handler = None
 
-    tagger.startTimer(1000)
-    sys.exit(tagger.run())
+    # No `else` statement is needed since pipe.Pipe has already sent picard_args to the existing instance
+    if should_start:
+        try:
+            from PyQt5.QtDBus import QDBusConnection
+            dbus = QDBusConnection.sessionBus()
+            dbus.registerService(PICARD_APP_ID)
+        except ImportError:
+            pass
+
+        tagger = Tagger(picard_args, unparsed_args, localedir, autoupdate, pipe_handler=pipe_handler)
+
+        # Initialize Qt default translations
+        translator = QtCore.QTranslator()
+        locale = QtCore.QLocale()
+        translation_path = QtCore.QLibraryInfo.location(QtCore.QLibraryInfo.LibraryLocation.TranslationsPath)
+        log.debug("Looking for Qt locale %s in %s", locale.name(), translation_path)
+        if translator.load(locale, "qtbase_", directory=translation_path):
+            tagger.installTranslator(translator)
+        else:
+            log.debug('Qt locale %s not available', locale.name())
+
+        tagger.startTimer(1000)
+        sys.exit(tagger.run())
+    else:
+        # just a custom exit code to show that picard instance wasn't created
+        sys.exit(30403)
```

Comment:
Instead of indenting the whole block, since we now exit in both cases, I suggest we use:

```python
if not should_start:
    # just a custom exit code to show that picard instance wasn't created
    sys.exit(30403)

try:
    from PyQt5.QtDBus import QDBusConnection
...
tagger.startTimer(1000)
sys.exit(tagger.run())
```

Reply:
Okay, looks fine to me